lib/olive_branch_patch.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# This will monkey patch olivebranch middleware https://github.com/vigetlabs/olive_branch/blob/master/lib/olive_branch/middleware.rb
+# so that when the VA inflection changes a key like something_va_something, the results will
+# have the old-inflection somethingVaSomething.
+
+module OliveBranchMiddlewareExtension
+  def call(env)
+    result = super(env)
+    _status, headers, response = result
+    # olive branch uses this private method to determine if a tranformation should happen https://github.com/vigetlabs/olive_branch/blob/master/lib/olive_branch/middleware.rb#L92
+    return result if send(:exclude_response?, env, headers)
+
+    if env['HTTP_X_KEY_INFLECTION'] =~ /camel/i
+      response.each do |json|
+        un_camel_va_keys!(json)
+      end
+    end
+    result
+  end
+
+  private
+
+  VA_KEY_REGEX = /("[^"]+VA[^"]*"):/.freeze
+
+  def un_camel_va_keys!(json)
+    # rubocop:disable Style/PerlBackrefs
+    # gsub with a block explicitly sets backrefs correctly https://ruby-doc.org/core-2.6.6/String.html#method-i-gsub
+    json.gsub!(VA_KEY_REGEX) do
+      key = $1
+      "#{key.gsub('VA', 'Va')}:"
+    end
+    # rubocop:enable Style/PerlBackrefs
+  end
+end
```

Comment:
Is there any way we could accomplish this without regexing unparsed json?

Reply:
I can't say "no".  But this seemed to be the most clear and straightforward way to accomplish transforming complex keys.  Is there a use-case it's breaking or a similar situation you need ideas for?

Reply:
It should seems a little dangerous when we could parse the json, recurse through it, and operate only on the keys. Agreed, with your regex, it's extremely unlikely that values will be affected, but still possible:
```
require 'json'

VA_KEY_REGEX = /("[^"]+VA[^"]*"):/.freeze

def un_camel_va_keys!(json)
  json.gsub!(VA_KEY_REGEX) do
    key = $1
    puts key
    "#{key.gsub('VA', 'Va')}:"
  end
end

json = <<-HEREDOC
{
  "disabilities": [
    "\\"CVA\\": stemming from traumatic brain injury."
  ]
}
HEREDOC

JSON.parse json

un_camel_va_keys! json

json
# => "{\n  \"disabilities\": [\n    \"\\\"CVa\\\": stemming from traumatic brain injury.\"\n  ]\n}\n"
```

Reply:
Is that a real value?  Are there instances of strings that include a quoted value, a colon, and more data?

Reply:
🤷‍♂️ lot's of strange data in the wild, but, like I said, I would guess extremely unlikely.
then again, murphy's law

Reply:
I feel that.  But if it's nothing something we'd expect, I like this approach.  Got to remember this code is run for EVERY response we generate; so parsing json, manipulating a hash, and converting it back to a string will take up time and electricity for a "what if?".